File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/generate/api/CodeGenerator.java
package io.github.potjerodekool.openapi.common.generate.api;

import io.github.potjerodekool.codegen.Environment;
import io.github.potjerodekool.openapi.common.ApiConfiguration;
import io.github.potjerodekool.openapi.common.OpenApiEnvironment;
import io.swagger.v3.oas.models.OpenAPI;

public interface CodeGenerator {
    void generateCommon(OpenApiEnvironment openApiEnvironment);

    void generateApi(OpenApiEnvironment openApiEnvironment,
                     OpenAPI openApi,
                     ApiConfiguration apiConfiguration,
                     boolean generateConfigs);
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/generate/api/CodeGenerator.kt
package io.github.potjerodekool.openapi.common.generate.api

import io.github.potjerodekool.openapi.common.ApiConfiguration
import io.github.potjerodekool.openapi.common.OpenApiEnvironment
import io.swagger.v3.oas.models.OpenAPI

interface CodeGenerator {
    fun generateCommon(openApiEnvironment: OpenApiEnvironment)

    fun generateApi(
        openApiEnvironment: OpenApiEnvironment,
        openApi: OpenAPI,
        apiConfiguration: ApiConfiguration,
        generateConfigs: Boolean
    )
}
